app/controllers/api/base_controller/parameters.rb

```diff
@@ -160,20 +160,24 @@ def sort_params(klass)
         orders = String(params['sort_order']).split(",")
         options = String(params['sort_options']).split(",")
         params['sort_by'].split(",").zip(orders).collect do |attr, order|
-          if klass.attribute_method?(attr) || klass.method_defined?(attr) || attr == klass.primary_key
-            sort_directive(attr, order, options)
+          if klass.virtual_attribute?(attr) && !klass.attribute_supported_by_sql?(attr)
```

Comment:
ok

I don't think this is necessary:
```ruby
klass.attribute_method?(attr) || klass.method_defined?(attr) || attr == klass.primary_key
```

@abellotti you ok just using `klass.attribute_supported_by_sql?(attr)` and dropping the ever neat previous one?

Did we need to support any ruby method or only fields?